### PR TITLE
Add new transform functions for converting to vehicle frame and padding.

### PR DESCRIPTION
**Pull request by Guowang**

### CODE REVIEW

File: keras_cv/layers/object_detection3d/utils.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+def HasRank(tensor, expected_rank):
```
Comment (ianstenbit) [inline keras_cv/layers/object_detection3d/utils.py:18]: Let's move these utils to keras_cv/utils/tensor_utils.py and use lowercase_underscore naming convention (to match other utils).

It seems like these methods aren't 3D OD specific, so a more generic codepath for them probably makes more sense

File: keras_cv/layers/object_detection3d/utils_test.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import tensorflow as tf
+
+from keras_cv.layers.object_detection3d.utils import PadOrTrimTo
+
+
+class PadOrTrimToTest(tf.test.TestCase):
+    """Tests for PadOrTrimTo, branched from https://github.com/tensorflow/lingvo/blob/master/lingvo/core/py_utils_test.py."""
+
+    def test2DConstantShapePad(self):
```
Comment (ianstenbit) [inline keras_cv/layers/object_detection3d/utils_test.py:23]: nit: lowercase_underscore naming for these test methods too

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you!

File: keras_cv/layers/object_detection3d/utils.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection3d/utils.py:18]: Is it possible to make this private utils for now, before we consider how to generalize it and move to keras_cv/utils?

File: keras_cv/datasets/waymo/transformer.py [hunk 3]
```diff
@@ -481,6 +482,26 @@ def _get_point_label(
     )
 
 
+def _inv_loc(rot: tf.Tensor, loc: tf.Tensor) -> tf.Tensor:
```
Comment (tanzhenyu) [inline keras_cv/datasets/waymo/transformer.py:485]: I think this function is already provided in voxel_utils.py, can you re-use that one?

**Review by tanzhenyu — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: keras_cv/datasets/waymo/transformer.py (hunk 3 above)
Comment (Guowang) [reply keras_cv/datasets/waymo/transformer.py:485]: Seems no functions for invert location? I can see get_yaw_rotation but not fit here.

File: keras_cv/layers/object_detection3d/utils.py (hunk 1 above)
Comment (Guowang) [reply keras_cv/layers/object_detection3d/utils.py]: Add '_' prefix to indicate it is internal and add comments to docstring.

File: keras_cv/layers/object_detection3d/utils_test.py (hunk 2 above)
Comment (Guowang) [reply keras_cv/layers/object_detection3d/utils_test.py]: done

File: keras_cv/datasets/waymo/transformer.py (hunk 3 above)
Comment (tanzhenyu) [reply keras_cv/datasets/waymo/transformer.py:485]: Oops, sorry, should be here:
https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/object_detection3d/center_net_label_encoder.py#L55
Can we extract this out into the voxel_utils.py?

**Review by tanzhenyu — APPROVED** [review]
LGTM after re-using `inv_loc`

File: keras_cv/datasets/waymo/transformer.py (hunk 3 above)
Comment (Guowang) [reply keras_cv/datasets/waymo/transformer.py]: Sure, also move internal utils to voxel_utils.

**Pull request merged**